File: back-end/src/main/java/com/example/backend/service/SchoolService.java
package com.example.backend.service;

import com.example.backend.dto.school.SchoolRequest;
import com.example.backend.entities.School;
import com.example.backend.service.base.BaseService;

import java.util.List;
import java.util.Optional;

public interface SchoolService extends BaseService<School> {
    School createSchool(SchoolRequest school);
    School updateSchool(Long id, SchoolRequest school);
    List<School> findByNameContaining(String name);
    List<School> findNearbySchools(Double latitude, Double longitude, Double radiusInMeters);
    long countActiveSchools();
    Optional<School> findByIdWithStudents(Long id);
    void addStudent(Long schoolId, Long studentId);
    void removeStudent(Long schoolId, Long studentId);
    void addRoute(Long schoolId, Long routeId);
    void removeRoute(Long schoolId, Long routeId);
    List<School> findSchoolsWithStats();
    long countActiveStudents(Long schoolId);
    long countActiveRoutes(Long schoolId);
    long countActiveDrivers(Long schoolId);
    long countActiveVehicles(Long schoolId);
} 

File: front-end/src/services/SchoolService.ts
import type { School } from "../core/entities/school.entity"
import type { CreateSchoolRequest, UpdateSchoolRequest } from "../core/dto/school.dto"

export interface SchoolService {
  findAll(): Promise<School[]>
  findById(id: number): Promise<School>
  create(schoolRequest: CreateSchoolRequest): Promise<School>
  update(id: number, schoolRequest: UpdateSchoolRequest): Promise<School>
  deleteSchool(id: number): Promise<void>
  findNearbySchools(latitude: number, longitude: number, radiusInMeters: number): Promise<School[]>
  getSchoolStatistics(id: number): Promise<any>
  findSchoolsByNameContaining(name: string): Promise<School[]>
  getSchoolWithStudents(id: number): Promise<School>
  addStudent(schoolId: number, studentId: number): Promise<void>
  removeStudent(schoolId: number, studentId: number): Promise<void>
  addRoute(schoolId: number, routeId: number): Promise<void>
  removeRoute(schoolId: number, routeId: number): Promise<void>
  getSchoolsWithStats(): Promise<School[]>
  countActiveSchools(): Promise<number>
  
  // These aliases are used in the UI components
  getAllSchools(): Promise<School[]>
  getSchoolById(id: number): Promise<School>
  createSchool(schoolData: any): Promise<School>
  updateSchool(id: number, schoolData: any): Promise<School>
}

